cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClientCredentials.java

```diff
@@ -32,24 +40,53 @@
 
     private final AuthenticationContextProvider authenticationContextProvider;
 
+    private final Tracer tracer;
+
     public AzureClientCredentials(AzureCredentialView credentialView, LogLevel logLevel, CBRefreshTokenClientProvider cbRefreshTokenClientProvider,
-                    AuthenticationContextProvider authenticationContextProvider) {
+            AuthenticationContextProvider authenticationContextProvider, Tracer tracer) {
         this.authenticationContextProvider = authenticationContextProvider;
         this.cbRefreshTokenClientProvider = cbRefreshTokenClientProvider;
         this.credentialView = credentialView;
         this.logLevel = logLevel;
+        this.tracer = tracer;
         azureClientCredentials = getAzureCredentials();
     }
 
     public Azure getAzure() {
         return Azure
                 .configure()
+                .withInterceptor(getTracingInterceptor())
                 .withProxyAuthenticator(new JavaNetAuthenticator())
                 .withLogLevel(logLevel)
                 .authenticate(azureClientCredentials)
                 .withSubscription(credentialView.getSubscriptionId());
     }
 
+    private Interceptor getTracingInterceptor() {
+        return chain -> {
+            Request request = chain.request();
+            String loggingContext = request.header("x-ms-logging-context").split(" ")[0];
+            Span span = tracer.buildSpan("[" + request.method() + "] " + loggingContext)
+                    .addReference(References.FOLLOWS_FROM, tracer.activeSpan() != null ? tracer.activeSpan().context() : null)
+                    .start();
+            span.setTag("component", "java-azure-sdk");
```

Comment:
I think these magic strings like `component`, `http.method`, `error` should come from `TracingUtil` or similar class, so when you add support to another sdk, etc we won't duplicate those.